### PR TITLE
Fixing delete channel logic

### PR DESCRIPTION
**Pull request by vishalnarkhede**
## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] Code changes are tested

## Changelog

- Once client receive `channel.deleted` or `notification.channel_deleted`, we remove it from `client.activeChannels`. But this prevents channel listeners to be called for that event. So we should only remove it from activeChannels after listeners have been fired.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -1230,6 +1230,8 @@ export class StreamChat<
     if (channel) {
       channel._callChannelListeners(event);
     }
+
+    postListenerCallbacks.forEach((c) => c());
```
Comment (mahboubii) [inline src/client.ts:1234]:
```suggestion
    postListenerCallbacks.forEach(c);
```

File: src/client.ts [hunk 2]
```diff
@@ -2063,6 +2073,10 @@ export class StreamChat<
     //                        we will replace it with `cid`
     for (const key in this.activeChannels) {
       const channel = this.activeChannels[key];
+      if (channel.disconnected) {
+        continue;
+      }
```
Comment (mahboubii) [inline src/client.ts:2078]: should we delete the channel in this case or let it be replaced later?

File: src/client.ts [hunk 3]
```diff
@@ -1458,10 +1461,17 @@ export class StreamChat<
...
-      this.activeChannels[event.cid]?._disconnect();
-      delete this.activeChannels[event.cid];
       client.state.deleteAllChannelReference(event.cid);
+      this.activeChannels[event.cid]?._disconnect();
+
+      postListenerCallbacks.push(() => {
+        if (!event.cid) return;
+
+        delete this.activeChannels[event.cid];
+      });
```
Comment (mahboubii) [inline src/client.ts:1471]:
```suggestion
      if (event.cid) 
            postListenerCallbacks.push(() => {
              delete this.activeChannels[event.cid];
            });
```

**Review by mahboubii — APPROVED** [review]

File: src/client.ts (hunk 1 above)
Comment (vishalnarkhede) [reply src/client.ts:1234]: As discussed, resolved

File: src/client.ts (hunk 3 above)
Comment (vishalnarkhede) [reply src/client.ts:1471]: We already have the check, this one is just to satisfy typescript!!

**Pull request merged**